maven-core/src/main/java/org/apache/maven/bridge/MavenRepositorySystem.java

```diff
@@ -573,15 +573,6 @@ else if ( Artifact.SCOPE_COMPILE.equals( scope ) && Artifact.SCOPE_COMPILE.equal
     // Code taken from LegacyRepositorySystem
     //
 
-    public ArtifactRepository createDefaultRemoteRepository( MavenExecutionRequest request )
```

Comment:
I don't know where this code is used, but it is public and I think this class is accessible by all plugins.

Reply:
True, I chose to use a highly invasive approach. One could use the same approach like in the original commits. Retain public method, but return either `null` or `InvalidRepositoryException`.  The contract does not describe any behavior.

Reply:
Ok, let's see what will happen

Reply:
So leave as-is?

Reply:
yes